cmd/overlay/main.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/zeebo/errs"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/cfgstruct"
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/process"
+	"storj.io/storj/pkg/provider"
+	proto "storj.io/storj/protos/overlay"
+	"fmt"
+)
+
+var (
+	mon     = monkit.Package()
+	Error   = errs.Class("overlay error")
+	rootCmd = &cobra.Command{
+		Use:   "overlay",
+		Short: "Overlay cache management",
+	}
+	addCmd = &cobra.Command{
+		Use:   "add",
+		Short: "Add nodes to the overlay cache",
+		RunE:  cmdAdd,
+	}
+	listCmd = &cobra.Command{
+		Use:   "list",
+		Short: "List nodes in the overlay cache",
+		RunE:  cmdList,
+	}
+
+	cacheCfg struct {
+		Identity provider.IdentityConfig
+		Kademlia kademlia.Config
+		cacheConfig
+	}
+
+	defaultConfDir = "$HOME/.storj/hc"
+)
+
+func init() {
+	rootCmd.AddCommand(addCmd)
+	rootCmd.AddCommand(listCmd)
+	cfgstruct.Bind(addCmd.Flags(), &cacheCfg)
+	cfgstruct.Bind(listCmd.Flags(), &cacheCfg)
+}
+
+func cmdList(cmd *cobra.Command, args []string) (err error) {
+	f := func(c *overlay.Cache) error {
+		keys, err := c.DB.List(nil, 0)
+		if err != nil {
+			return err
+		}
+
+		for _, k := range keys {
+			n, err := c.Get(process.Ctx(cmd), string(k))
+			if err != nil {
+				fmt.Printf("ID: %s; error getting value\n", k)
+				// return Error.Wrap(err)
+			}
+			if n != nil {
+				fmt.Printf("ID: %s; Address: %s\n", k, n.Address.Address)
+				continue
+			}
+			fmt.Printf("ID: %s; Address: nil\n", k)
+		}
+
+		return nil
+	}
+	c := cacheInjector{
+		cacheConfig: cacheCfg.cacheConfig,
+		c:           f,
+	}
+
+	return cacheCfg.Identity.Run(process.Ctx(cmd),
```

Comment:
you don't need the identity stuff, right? you just need to make an *overlay.Cache?